src/upstream_patches_ui.sh

```diff
@@ -432,6 +462,34 @@ function show_series_details()
               create_message_box 'Error' 'Could not download patch(es):'$'\n'"- ${series['patch_title']}"$'\n'"[error message] ${output}"
             fi
             ;;
+          'Apply')
+            if [[ -z "${lore_config['kernel_tree_path']}" || -z "${lore_config['kernel_tree_branch']}" ]]; then
+              create_message_box 'Error' 'The "Kernel Tree Path" and "Kernel Tree Branch" configs must be set.'$'\n''Go to Dashboard -> Settings.'
+              return
+            fi
+            message_box="The branch '${lore_config['kernel_tree_branch']}' of the Linux kernel tree located in '${lore_config['kernel_tree_path']}' "
+            message_box+='will be updated and a new branch based on it will be created for the patch application. Do you wish to proceed?'
+            create_yes_no_prompt 'Warning' "$message_box"
+            case "$?" in
+              0) # Yes
+                create_loading_screen_notification 'Applying patch(es)'$'\n'"- ${series['patch_title']}"
+                output=$(download_series "${series['patch_url']}" "${lore_config['save_patches_to']}")
+                if [[ "$?" != 0 ]]; then
+                  create_message_box 'Error' 'Could not download patch(es):'$'\n'"- ${series['patch_title']}"$'\n'"[error message] ${output}"
+                  return
+                fi
+                output=$(apply_patchset "${series['patch_title']}" "$output" "${lore_config['kernel_tree_path']}" "${lore_config['kernel_tree_branch']}")
+                if [[ "$?" != 0 ]]; then
+                  create_message_box 'Fail' 'Could not apply patch(es):'$'\n'"- ${series['patch_title']}"$'\n'$'\n'"[error message] ${output}"
+                  return
+                fi
+                create_message_box 'Success' "'${series['patch_title']}' applied."$'\n'"${output}"
+                ;;
+              1) # No
+                create_message_box '' 'Patch application aborted!'
+                ;;
+            esac
```

Comment:
Is this the same code as the previous one? Can you create a common function for that?

Reply:
Nice catch, this is also really bothering me as it is happening throughout the whole file. Going to work on that in a separate PR for refactoring this whole file.